sourcecode/chesssimul/board.py

```diff
@@ -1,169 +1,182 @@
+    to_move: Color
+    can_white_queen_side_castle: bool
+    can_white_king_side_castle: bool
+    can_black_queen_side_castle: bool
+    can_black_king_side_castle: bool
+    en_passant_target_square: Square
+
+    def getSquare(self, square: str) -> Square:
+        square_list = list(square)
+        return self.board[Lines.get(square_list[1])][Columns.get(square_list[0])]
+
+    def init_board(self):
+        self.board = list()
+        for lin in range(8):
+            self.board.append(list())
+            for col in range(8):
+                self.board[lin].append(Square())
+
+        for col in range(8):
+            self.board[Lines.get('7')][col].piece = str_to_piece['P']
+            self.board[Lines.get('2')][col].piece = str_to_piece['p']
+        self.getSquare('a1').piece = str_to_piece['r']
+        self.getSquare('b1').piece = str_to_piece['n']
+        self.getSquare('c1').piece = str_to_piece['b']
+        self.getSquare('d1').piece = str_to_piece['q']
+        self.getSquare('e1').piece = str_to_piece['k']
+        self.getSquare('f1').piece = str_to_piece['b']
+        self.getSquare('g1').piece = str_to_piece['n']
+        self.getSquare('h1').piece = str_to_piece['r']
+        self.getSquare('a8').piece = str_to_piece['R']
+        self.getSquare('b8').piece = str_to_piece['N']
+        self.getSquare('c8').piece = str_to_piece['B']
+        self.getSquare('d8').piece = str_to_piece['Q']
+        self.getSquare('e8').piece = str_to_piece['K']
+        self.getSquare('f8').piece = str_to_piece['B']
+        self.getSquare('g8').piece = str_to_piece['N']
+        self.getSquare('h8').piece = str_to_piece['R']
+
+    def to_unicode(self) -> str:
+        board_string = ''
+        for line in range(8):
+            board_string += '|'
+            for col in range(8):
+                square = self.board[line][col]
+                if square.isEmpty():
+                    board_string += ' '
+                else:
+                    board_string += square.piece.to_unicode()
+                board_string += '|'
+            board_string += '\n'
+        return board_string
+
+
+    def make_move(self, move: Move):
+        print('make move')
+        #postion de la pièce avant son déplacement dans l'échiquier
+        start_square: Square = self.board[move.start[0]][move.start[1]]
+        #Y a t'il une pièce à déplacer
+        if not (start_square.isEmpty()):
+            #type de la pièce que nous voulons déplacer
+            type_piece_start: PieceType = start_square.piece.kind
+            print(type_piece_start)
+            # Si c'est le cavalier
+            if (type_piece_start == PieceType.KNIGHT):
+                print("vérification du coup du cavalier")
+                self.move_knight(move)
+            if (type_piece_start == PieceType.BISHOP):
+                print("vérification du coup du fou")
+                self.move_bishop(move)
+
+
+
+    def move_knight(self, move: Move):

```

Comment:
I think the function is missing a check to see if the end square coordinates are inside the board.

Reply:
ditto return boolean

Reply:
You also need to think about the implementation of a verification to see wether : 
-The king is currently in check, meaning that any move that does not stop the check is illegal
-Moving that piece doesn't leave the king is check (i.e, check that this piece is not [pinned](https://en.wikipedia.org/wiki/Pin_(chess)))